src/protocols/Protocols.h

```diff
@@ -92,10 +93,49 @@ CHIP_STANDARD_PROTOCOL(Echo, 0x0004)                      // Echo Protocol.  To
 // Protocols reserved for internal protocol use
 static constexpr Id NotSpecified(VendorId::NotSpecified, 0xFFFF); // The profile ID is either not specified or a wildcard
 
-// Pre-delare our MessageTypeTraits so message type headers know what they are
-// specializing.
+//
+// Pre-declaration of type traits that protocol headers are expected to define.
+//
+// A protocol header should first define an enumeration that lists the various message types as enum classes.
+//
+// It should then define MessageTypeTraits as a template specialization of that enumeration containing two methods:
+//      1. static constexpr const Protocols::Id & ProtocolId() that returns the Protocol ID
+//      2. static auto GetTypeToNameTable() that returns a std::array<MessageTypeNameLookup, N> where N = number of messages in
+//      protocol.
+//
 template <typename T>
 struct MessageTypeTraits;
 
+//
+// Encapsulates a tuple of message type ID and its associated name.
+//
+struct MessageTypeNameLookup
+{
+    template <typename T>
```

Comment:
Having this above the comment like this is pretty confusing to read.  Could it please be moved to right above the `constexpr, below the comment?